<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>571</width>
    <height>563</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>571</width>
    <height>563</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>571</width>
    <height>563</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>546</width>
      <height>501</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayoutTabButton">
     <property name="spacing">
      <number>6</number>
     </property>
     <item>
      <widget class="QTabWidget" name="tabWidget">
       <property name="minimumSize">
        <size>
         <width>544</width>
         <height>470</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>544</width>
         <height>470</height>
        </size>
       </property>
       <property name="currentIndex">
        <number>5</number>
       </property>
       <widget class="QWidget" name="instance">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <attribute name="title">
         <string>Instance</string>
        </attribute>
        <widget class="QWidget" name="layoutWidget">
         <property name="geometry">
          <rect>
           <x>9</x>
           <y>9</y>
           <width>501</width>
           <height>101</height>
          </rect>
         </property>
         <layout class="QVBoxLayout" name="verticalLayoutInstance">
          <item>
           <widget class="QLabel" name="applicationNameLabel">
            <property name="text">
             <string>Application Name:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="applicationNameInput">
            <property name="placeholderText">
             <string>Enter your application name here</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="showValidationLayerDebugInfoCheckBox">
            <property name="text">
             <string>Show Validation Layer Debug Info</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="runOnMacosCheckBox">
            <property name="text">
             <string>Run on MacOS</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
       <widget class="QWidget" name="physicalDevice">
        <attribute name="title">
         <string>Physical Device</string>
        </attribute>
        <widget class="QWidget" name="layoutWidget">
         <property name="geometry">
          <rect>
           <x>9</x>
           <y>9</y>
           <width>521</width>
           <height>111</height>
          </rect>
         </property>
         <layout class="QVBoxLayout" name="verticalLayoutPhysicalDevice">
          <item>
           <widget class="QCheckBox" name="chooseGPUOnStartupCheckBox">
            <property name="text">
             <string>Choose GPU on Application start</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="saveEnergyForMobileCheckBox">
            <property name="text">
             <string>Save Energy for Mobile</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
       <widget class="QWidget" name="logicalDevice">
        <attribute name="title">
         <string>Logical Device</string>
        </attribute>
        <widget class="QWidget" name="layoutWidget">
         <property name="geometry">
          <rect>
           <x>9</x>
           <y>9</y>
           <width>521</width>
           <height>94</height>
          </rect>
         </property>
         <layout class="QVBoxLayout" name="verticalLayoutLogicalDevice">
          <item>
           <widget class="QLabel" name="deviceExtensionsLabel">
            <property name="text">
             <string>Device Extensions:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QListWidget" name="deviceExtensionsList">
            <item>
             <property name="text">
              <string>VK_KHR_SWAPCHAIN_EXTENSION_NAME</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>other extensions</string>
             </property>
             <property name="flags">
              <set>ItemIsSelectable|ItemIsDragEnabled|ItemIsUserCheckable</set>
             </property>
            </item>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
       <widget class="QWidget" name="swapchain">
        <attribute name="title">
         <string>Swapchain</string>
        </attribute>
        <widget class="QWidget" name="layoutWidget">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>10</y>
           <width>531</width>
           <height>431</height>
          </rect>
         </property>
         <layout class="QVBoxLayout" name="verticalLayoutSwapChain">
          <item>
           <layout class="QVBoxLayout" name="verticalLayoutImageDImensions">
            <item>
             <widget class="QLabel" name="imageDimensionsLabel">
              <property name="text">
               <string>Image Dimensions:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSplitter" name="imageDimensionsSplitter">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <widget class="QWidget" name="layoutWidget">
               <layout class="QHBoxLayout" name="horizontalLayoutImageHeight">
                <item>
                 <widget class="QLabel" name="imageHeightLabel">
                  <property name="text">
                   <string>Height:</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QSpinBox" name="imageHeightInput"/>
                </item>
               </layout>
              </widget>
              <widget class="QWidget" name="layoutWidget">
               <layout class="QHBoxLayout" name="horizontalLayoutImageWidth">
                <item>
                 <widget class="QLabel" name="imageWidthLabel">
                  <property name="text">
                   <string>Width:</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QSpinBox" name="imageWidthInput"/>
                </item>
               </layout>
              </widget>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QCheckBox" name="lockWindowSizeCheckBox">
            <property name="text">
             <string>Lock Window size</string>
            </property>
            <property name="tristate">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayoutImageClearColor">
            <item>
             <widget class="QLabel" name="imageClearColorLabel">
              <property name="text">
               <string>Image Clear Color:</string>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayoutRGBA">
              <item>
               <layout class="QHBoxLayout" name="horizontalLayoutClearColorR">
                <item>
                 <widget class="QLabel" name="clearColorRLabel">
                  <property name="text">
                   <string>R</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QDoubleSpinBox" name="clearColorRInput">
                  <property name="maximum">
                   <double>1.000000000000000</double>
                  </property>
                  <property name="singleStep">
                   <double>0.010000000000000</double>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayoutClearColorG">
                <item>
                 <widget class="QLabel" name="clearColorGLabel">
                  <property name="text">
                   <string>G</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QDoubleSpinBox" name="clearColorGInput">
                  <property name="maximum">
                   <double>1.000000000000000</double>
                  </property>
                  <property name="singleStep">
                   <double>0.010000000000000</double>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayoutClearColorB">
                <item>
                 <widget class="QLabel" name="clearColorBLabel">
                  <property name="text">
                   <string>B</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QDoubleSpinBox" name="clearColorBInput">
                  <property name="maximum">
                   <double>1.000000000000000</double>
                  </property>
                  <property name="singleStep">
                   <double>0.010000000000000</double>
                  </property>
                  <property name="value">
                   <double>0.000000000000000</double>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayoutClearColorA">
                <item>
                 <widget class="QLabel" name="clearColorALabel">
                  <property name="text">
                   <string>A</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QDoubleSpinBox" name="clearColorAInput">
                  <property name="maximum">
                   <double>1.000000000000000</double>
                  </property>
                  <property name="singleStep">
                   <double>0.010000000000000</double>
                  </property>
                  <property name="value">
                   <double>1.000000000000000</double>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayoutFramesInFlight">
            <item>
             <widget class="QLabel" name="framesInFlightLabel">
              <property name="text">
               <string>Frames in Flight</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSpinBox" name="framesInFlightInput">
              <property name="maximum">
               <number>999</number>
              </property>
              <property name="value">
               <number>2</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayoutImageSelectors">
            <item>
             <layout class="QVBoxLayout" name="verticalLayoutImageSelectorsLabel">
              <item>
               <widget class="QLabel" name="ImageUsageLabel">
                <property name="text">
                 <string>Image Usage</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="imagePresentationModeLabel">
                <property name="text">
                 <string>Presentation Mode</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="imageFormatLabel">
                <property name="text">
                 <string>Image Format</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="imageColorSpaceLabel">
                <property name="text">
                 <string>Image Color Space</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QVBoxLayout" name="verticalLayoutImageSelectorsBox">
              <item>
               <widget class="QComboBox" name="imageUsageInput">
                <property name="minimumSize">
                 <size>
                  <width>251</width>
                  <height>0</height>
                 </size>
                </property>
                <property name="editable">
                 <bool>false</bool>
                </property>
                <property name="currentText">
                 <string notr="true">VK_IMAGE_USAGE_COLOR_ATTACHMENT_BIT</string>
                </property>
                <item>
                 <property name="text">
                  <string>VK_IMAGE_USAGE_COLOR_ATTACHMENT_BIT</string>
                 </property>
                </item>
               </widget>
              </item>
              <item>
               <widget class="QComboBox" name="presentationModeInput">
                <property name="minimumSize">
                 <size>
                  <width>231</width>
                  <height>0</height>
                 </size>
                </property>
                <property name="editable">
                 <bool>false</bool>
                </property>
                <property name="currentText">
                 <string notr="true">VK_PRESENT_MODE_MAILBOX_KHR</string>
                </property>
                <item>
                 <property name="text">
                  <string>VK_PRESENT_MODE_MAILBOX_KHR</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>VK_PRESENT_MODE_IMMEDIATE_KHR</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>VK_PRESENT_MODE_FIFO_KHR</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>VK_PRESENT_MODE_FIFO_RELAXED_KHR</string>
                 </property>
                </item>
               </widget>
              </item>
              <item>
               <widget class="QComboBox" name="imageFormatInput">
                <property name="minimumSize">
                 <size>
                  <width>231</width>
                  <height>0</height>
                 </size>
                </property>
                <property name="editable">
                 <bool>false</bool>
                </property>
                <property name="currentText">
                 <string notr="true">VK_FORMAT_B8G8R8A8_SRGB</string>
                </property>
                <item>
                 <property name="text">
                  <string>VK_FORMAT_B8G8R8A8_SRGB</string>
                 </property>
                </item>
               </widget>
              </item>
              <item>
               <widget class="QComboBox" name="imageColorSpaceInput">
                <property name="minimumSize">
                 <size>
                  <width>231</width>
                  <height>0</height>
                 </size>
                </property>
                <property name="editable">
                 <bool>false</bool>
                </property>
                <property name="currentText">
                 <string notr="true">VK_COLORSPACE_SRGB_NONLINEAR_KHR</string>
                </property>
                <item>
                 <property name="text">
                  <string>VK_COLORSPACE_SRGB_NONLINEAR_KHR</string>
                 </property>
                </item>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </widget>
       <widget class="QWidget" name="shader">
        <attribute name="title">
         <string>Shader</string>
        </attribute>
        <widget class="QWidget" name="">
         <property name="geometry">
          <rect>
           <x>12</x>
           <y>12</y>
           <width>514</width>
           <height>431</height>
          </rect>
         </property>
         <layout class="QVBoxLayout" name="verticalLayoutShader">
          <item>
           <layout class="QGridLayout" name="gridLayoutShaderFiles">
            <item row="0" column="0">
             <widget class="QLabel" name="vertexShaderFileLabel">
              <property name="text">
               <string>Vertex Shader File</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QLineEdit" name="vertexShaderFileInput">
              <property name="minimumSize">
               <size>
                <width>371</width>
                <height>0</height>
               </size>
              </property>
             </widget>
            </item>
            <item row="0" column="2">
             <widget class="QToolButton" name="vertexShaderFileToolButton">
              <property name="text">
               <string>...</string>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="fragmentShaderFileLabel">
              <property name="text">
               <string>Fragment Shader File</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QLineEdit" name="fragmentShaderFileInput">
              <property name="minimumSize">
               <size>
                <width>371</width>
                <height>0</height>
               </size>
              </property>
             </widget>
            </item>
            <item row="1" column="2">
             <widget class="QToolButton" name="fragmentShaderFileToolButton">
              <property name="text">
               <string>...</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayoutShaderEntryFunction">
            <item>
             <widget class="QLabel" name="shaderEntryFunctionNameLabel">
              <property name="text">
               <string>Shader Entry Function Name</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="shaderEntryFunctionNameInput">
              <property name="font">
               <font>
                <kerning>true</kerning>
               </font>
              </property>
              <property name="alignment">
               <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
              </property>
              <property name="placeholderText">
               <string>main</string>
              </property>
              <property name="clearButtonEnabled">
               <bool>false</bool>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayoutVertexTopology">
            <item>
             <widget class="QLabel" name="vertexTopologyLabel">
              <property name="text">
               <string>Vertex Topology</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="vertexTopologyInput">
              <property name="minimumSize">
               <size>
                <width>341</width>
                <height>0</height>
               </size>
              </property>
              <item>
               <property name="text">
                <string>VK_PRIMITIVE_TOPOLOGY_TRIANGLE_LIST</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>VK_PRIMITIVE_TOPOLOGY_POINT_LIST</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>VK_PRIMITIVE_TOPOLOGY_LINE_LIST</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>VK_PRIMITIVE_TOPOLOGY_TRIANGLE_STRIP</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>VK_PRIMITIVE_TOPOLOGY_TRIANGLE_FAN</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>VK_PRIMITIVE_TOPOLOGY_TRIANGLE_LIST_WITH_ADJACENCY</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>VK_PRIMITIVE_TOPOLOGY_TRIANGLE_STRIP_WITH_ADJACENCY</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>VK_PRIMITIVE_TOPOLOGY_LINE_STRIP</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>VK_PRIMITIVE_TOPOLOGY_LINE_LIST_WITH_ADJACENCY</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>VK_PRIMITIVE_TOPOLOGY_LINE_STRIP_WITH_ADJACENCY</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>VK_PRIMITIVE_TOPOLOGY_PATCH_LIST</string>
               </property>
              </item>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QCheckBox" name="useIndexedVerticesCheckBox">
            <property name="text">
             <string>Use Indexed Vertices</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="reduceSpirvCodeSizeCheckBox">
            <property name="text">
             <string>Reduce SPIR-V Code Size</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
       <widget class="QWidget" name="model">
        <attribute name="title">
         <string>Model</string>
        </attribute>
       </widget>
       <widget class="QWidget" name="graphicsPipeline">
        <attribute name="title">
         <string>Graphics Pipeline</string>
        </attribute>
        <widget class="QLabel" name="rasterizerLabel">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>10</y>
           <width>51</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>Rasterizer</string>
         </property>
        </widget>
        <widget class="Line" name="rasterizerSpacer">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>50</y>
           <width>761</width>
           <height>20</height>
          </rect>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </widget>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="generateVulkanCodeButton">
       <property name="text">
        <string>Generate Vulkan Code</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>571</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionNew_File"/>
    <addaction name="actionLoad_from_File"/>
    <addaction name="actionSave_to_File"/>
    <addaction name="separator"/>
    <addaction name="actionClose"/>
   </widget>
   <widget class="QMenu" name="menuExport">
    <property name="title">
     <string>Generate</string>
    </property>
    <addaction name="actionExport_to_Vulkan"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuExport"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionLoad_from_File">
   <property name="text">
    <string>Load from File</string>
   </property>
  </action>
  <action name="actionSave_to_File">
   <property name="text">
    <string>Save to File</string>
   </property>
  </action>
  <action name="actionClose">
   <property name="text">
    <string>Close</string>
   </property>
  </action>
  <action name="actionExport_to_Vulkan">
   <property name="text">
    <string>Export to Vulkan</string>
   </property>
  </action>
  <action name="actionNew_File">
   <property name="text">
    <string>New File</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
